pkg/apis/core/validation/validation.go

```diff
@@ -2966,7 +2978,7 @@ func validatePodDNSConfig(dnsConfig *core.PodDNSConfig, dnsPolicy *core.DNSPolic
 			if strings.HasSuffix(search, ".") {
 				search = search[0 : len(search)-1]
 			}
-			allErrs = append(allErrs, ValidateDNS1123Subdomain(search, fldPath.Child("searches").Index(i))...)
+			allErrs = append(allErrs, validateDomainName(search, fldPath.Child("searches").Index(i))...)
```

Comment:
We need to distinguish updates of existing values from net-new usage.  For example, if I upgrade my cluster, create a deployment using a previously-invalid value, then downgrade, I should still be able to update my deployment.  But I should not be able to make a new deployment using invalid values unless the gate is enabled.

Reply:
@thockin  slightly unrelated to the PR, just for learning purposes, how does this work for skewed apiservers?
apiserver A with the feature gate enabled and new version will allow to use the new search domain "my-domain.local", but clients using apiserver B with version-1 will fail, right?

Reply:
I've found the answer, https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#changing-the-api
just for reference, 

> The recommended place to do this is in the REST storage strategy's PrepareForCreate/PrepareForUpdate methods: